### PR TITLE
fix: Correctly log request and response for HTTP in Elasticsearch.

### PR DESCRIPTION
**Pull request by jwalton**
A bug was making it so we were rarely logging `req` and `res` to Elasticsearch.  Worse, sometimes we were overwriting `req` with a string; this makes it so if we now try to set `req` to an object,
Elasticsearch will reject the log entry.  This fixes these problems, and also remaps `req` to
`request` and `res` to `response` to avoid the existing `req` field.

Fixes benbria/loop#6860.

### CODE REVIEW

File: src/winston/index.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export function connectToElasticsearch() {
                     clientOpts: { hosts: hosts },
                     indexPrefix: "loop",
                     format: winston.format.combine(
-                        requestFormat({debug: true}),
+                        requestFormat({debug: false}),
```
Comment (jwalton) [inline src/winston/index.ts]: This is the really important part of the fix.

File: src/winston/format/requestFormat.ts [hunk 2]
```diff
@@ -72,22 +72,21 @@ const requestFormat : logform.FormatWrap = format((
...
+        const reqString = requestToString(request, response, info.user, options.useColors);
 
         // If there's no message, then replace the message with the request
         if(!info.message) {
-            info.message = request;
-        } else {
-            // Force `req` to be a string.
-            (info as any).req = request;
+            info.message = reqString;
+        } else if(options.debug) {
```
Comment (jwalton) [inline src/winston/format/requestFormat.ts]: Adding `if(options.debug)` here is pretty important, too - this is supposed to give us a single line `req` in debug mode, but it was actually overwriting `req` with a string.

**Comment by macpham** [comment]
In the existing logs the `req` and `res` were mostly missing? And after this change, it should come in as `request` and `response`?

**Comment by jwalton** [comment]
@macpham Yup.  `res` is always missing, `req` is sometimes there (only when there's a `req` and a message) but it's a string.

**Pull request merged**